virtdata-lib-basics/src/main/java/io/nosqlbench/virtdata/library/basics/shared/vectors/dnn/DNN_angular1_neighbors.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2023 nosqlbench
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.nosqlbench.virtdata.library.basics.shared.vectors.dnn;
+
+import io.nosqlbench.virtdata.api.annotations.Categories;
+import io.nosqlbench.virtdata.api.annotations.Category;
+import io.nosqlbench.virtdata.api.annotations.ThreadSafeMapper;
+
+import java.util.function.IntFunction;
+
+/**
+ * Compute the indices of the neighbors of a given v using DNN mapping.
+ * To avoid ambiguity on equidistant neighbors, odd neighborhood sizes are preferred.
+ */
+@ThreadSafeMapper
+@Categories(Category.experimental)
+public class DNN_angular1_neighbors implements IntFunction<int[]> {
```

Comment:
We may need another version of this which supports LongFunction<int[]> so that there is a native cycle-> binding type. I think it should work already as long as there is already a LongFunction -> IntFunction adapter, but since these are type (and generic type) specific, it may just be easier to add a LongFunction variant which simply calls into this one.